### PR TITLE
refactor: make utils flatten

### PR DESCRIPTION
**Pull request by homura**
# What Changed

- flatted `utils` package
- correct `numeral` dep
- update deps

## Motivation

Before [publishing](https://github.com/ckb-js/nexus/issues/191) `utils` to npm registry, we should make the exported function be stable, so I have 

- removed unnecessary deps(`lumos/codec`) from utils to keep dependencies minimal
- flatted exported function to keep API clear

## Change Type

Indicate the type of change your pull request is:

- [ ] `documentation`
- [ ] `patch`
- [ ] `minor`
- [ ] `major`

### CODE REVIEW

File: packages/types/src/base.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 export type Promisable<T> = PromiseLike<T> | T;
 
+export type Resolvable<T> = T | PromiseLike<T> | (() => T) | (() => PromiseLike<T>);
```
Comment (zhangyouxin) [inline packages/types/src/base.ts:3]:
```suggestion
export type Resolvable<T> = Promisable<T> | (() => Promisable<T>);
```

File: packages/utils/__tests__/asserts.ts [hunk 2]
```diff
@@ -1,12 +1,6 @@
-import { asserts } from '../src';
+import { assert } from '../src';
 
 it('utils#asserts', () => {
-  expect(() => asserts.nonFalsy(0)).toThrow();
-  expect(() => asserts.nonFalsy('')).toThrow();
-  expect(() => asserts.nonFalsy(undefined)).toThrow();
-  expect(() => asserts.nonFalsy(null)).toThrow();
-
-  expect(() => asserts.nonFalsy(1)).not.toThrow();
```
Comment (zhangyouxin) [inline packages/utils/__tests__/asserts.ts:9]: I would replace with corresponding test cases

```
  expect(() => assert(0)).toThrow();
  expect(() => assert(1)).not.toThrow();
```

Comment (homura) [reply packages/utils/__tests__/asserts.ts:9]: supplied

File: packages/types/src/base.ts (hunk 1 above)
Comment (homura) [reply packages/types/src/base.ts]: refactored

File: packages/utils/src/asserts.ts [hunk 3]
```diff
@@ -1,15 +1,26 @@
...
+/**
+ * Assert condition is truthy, otherwise throw error.
+ * This function usually used with TypeScript's `asserts` type guard
+ * @example
+ *   assert(typeof value === 'string', 'value must be a string')
+ *   assert(typeof value === 'string', 'value must be a string, got %s', typeof value)
+ * @param condition
+ * @param args
+ */
+export function assert(condition: unknown, ...args: unknown[]): asserts condition {
```
Comment (IronLu233) [inline packages/utils/src/asserts.ts:13]: Why not consider the implementation from community? It is more well-tested.

https://www.npmjs.com/package/assert

File: packages/utils/src/internal/format.ts [hunk 4]
```diff
@@ -1,9 +1,10 @@
-import type { Provider } from '@nexus-wallet/types/lib/services/common';
-import type { Promisable } from '@nexus-wallet/types';
-import { bytes, BytesLike } from '@ckb-lumos/codec';
```
Comment (IronLu233) [inline packages/utils/src/internal/format.ts:3]: Seems implements bytes.hexify twice... Is it violate "Do not repeat your self" principle?

File: packages/utils/src/asserts.ts (hunk 3 above)
Comment (homura) [reply packages/utils/src/asserts.ts:13]: 1. keep the error message style consistent, e.g. "Assertion failed: 1 + 1 should be 2, but got 3  (version=0.0.1)"
2. it is a simple function, there is no need to introduce a library for a function

File: packages/utils/src/internal/format.ts (hunk 4 above)
Comment (homura) [reply packages/utils/src/internal/format.ts:3]: only hexify for `Uint8Array` here, it is no need to import the entire `codec` library

**Review by zhangyouxin — APPROVED** [review]

**Review by IronLu233 — APPROVED** [review]

**Pull request merged**